application/apps/indexer/parsers/src/dlt/mod.rs

```diff
@@ -67,10 +70,11 @@ impl LogMessage for RawMessage {
 #[derive(Default)]
 pub struct DltParser<'m> {
     pub filter_config: Option<ProcessedDltFilterConfig>,
-    pub fibex_metadata: Option<&'m FibexMetadata>,
+    pub fibex_dlt_metadata: Option<&'m FibexDltMetadata>,
     pub fmt_options: Option<&'m FormatOptions>,
     pub with_storage_header: bool,
     ft_scanner: FtScanner,
+    fibex_someip_metadata: Option<&'m FibexSomeipMetadata>,
```

Comment:
"I think the main point is to replace the two optional fibex meta data with enum since they can't be provided at the same time." 
=> This assumption is not correct, they could be present same time!

In case the Fibex does contain metadata about NonVerbose DLT messages and SomeIp Services, both metatdata items would be used to to decode a DLT file containing any NonVerbose DLT messages or SomeIp payload in DLT Network-Trace messages.